internal/testutil/sqlite_testdb.go

```diff
@@ -0,0 +1,32 @@
+package testutil
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"gorm.io/driver/sqlite"
+	"gorm.io/gorm"
+)
+
+const SQLiteInMemoryPattern = "file:%s?mode=memory&cache=shared"
+
+func NewSQLiteTestDB(t *testing.T, cfg *gorm.Config) *gorm.DB {
+	t.Helper()
+
+	if cfg == nil {
+		cfg = &gorm.Config{}
+	}
+
+	// Use the test name plus a timestamp to generate a unique in-memory
+	// database per test
+	safeName := strings.ReplaceAll(t.Name(), "/", "_")
+	dsn := fmt.Sprintf(SQLiteInMemoryPattern, fmt.Sprintf("%s_%d", safeName, time.Now().UnixNano()))
+
+	db, err := gorm.Open(sqlite.Open(dsn), cfg)
```

Comment:
FWIW [:memory:](https://sqlite.org/inmemorydb.html) creates memory-only sqlite - unless you want to inspect the contents later it might be easier than crafting a filename. 
Otherwise https://godocs.io/os#CreateTemp could be an alternative for a temporal file name